### PR TITLE
Added http example

### PR DESCRIPTION
**Pull request by gkbrk**

### CODE REVIEW

File: examples/http.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                        _ => "Cannot find page\n"
+                    };
+                   
+                    write!(stream, "\
+                    HTTP/1.1 200 OK\r\n\
+                    Server: May\r\n\
+                    Content-Length: {}\r\n\
+                    Date: 1-1-2000\r\n\
+                    \r\n\
+                    {}", response.len(), response);
```
Comment (Xudong-Huang) [inline examples/http.rs]: directly call the `write!` macro on a raw writer would be very slow. I think internally it calls too much system write. change the code as following gives me a shock about wrk bench result
```rust
                    let s = format!("\
                        HTTP/1.1 200 OK\r\n\
                        Server: May\r\n\
                        Content-Length: {}\r\n\
                        Date: 1-1-2000\r\n\
                        \r\n\
                        {}", response.len(), response);

                    stream.write_all(s.as_bytes());
```

**Comment by Xudong-Huang** [comment]
Great work!
here is the result after the minor change
```sh
$ wrk http://127.0.0.1:8080 -d 10 -t 2 -c 200
Running 10s test @ http://127.0.0.1:8080
  2 threads and 200 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   513.65us  390.06us  11.29ms   98.38%
    Req/Sec   195.35k    14.04k  211.33k    87.50%
  3887636 requests in 10.03s, 344.80MB read
Requests/sec: 387632.18
Transfer/sec:     34.38MB
wrk http://127.0.0.1:8080 -d 10 -t 2 -c 200  3.46s user 16.24s system 196% cpu 10.039 tota
```
this is fastest wrk bench I have ever seen on my machine!
👍

File: examples/http.rs (hunk 1 above)
Comment (Xudong-Huang) [reply examples/http.rs]: by the way, you can use `std::io::BufWriter`  for internal cache things.

File: examples/http.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+extern crate may;
+extern crate httparse;
+extern crate bytes;
+
+use may::coroutine;
+use may::net::TcpListener;
+use httparse::Request;
```
Comment (Xudong-Huang) [inline examples/http.rs]: this is unused, remove it

File: examples/http.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    HTTP/1.1 200 OK\r\n\
+                    Server: May\r\n\
+                    Content-Length: {}\r\n\
+                    Date: 1-1-2000\r\n\
+                    \r\n\
+                    {}", response.len(), response);
+                    
+                    buf = buf.split_off(i);
+                } else {
+                    let mut tempBuf = vec![0; 512];
```
Comment (Xudong-Huang) [inline examples/http.rs]: rename to temp_buf, remove variable name warnings.

Comment (Xudong-Huang) [reply examples/http.rs]: need to remove all the compile warnings

**Comment by gkbrk** [comment]
@Xudong-Huang The things you mentioned should be good now. The result I get from wrk improved dramatically on my machine as well.

**Comment by Xudong-Huang** [comment]
thanks for the update. 🍻

**Pull request merged**